gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/ScalableGeneratorConnector.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.swe.runner.app.dichotomy.shift;
+
+import com.farao_community.farao.commons.EICode;
+import com.farao_community.farao.dichotomy.api.exceptions.ShiftingException;
+import com.powsybl.glsk.commons.ZonalData;
+import com.powsybl.iidm.modification.scalable.Scalable;
+import com.powsybl.iidm.network.*;
+import org.jgrapht.alg.util.Pair;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Before scaling: fetches scalable generators in ES and PT scalables. For generators connected to the grid through a
+ * transformer, if the transformer is disconnected, connects the transformer.
+ * After scaling: reverts unnecessary changes.
+ *
+ * @author Peter Mitri {@literal <peter.mitri at rte-france.com>}
+ */
+public class ScalableGeneratorConnector {
+    private Map<String, GeneratorState> changedGeneratorsInitialState = new HashMap<>();
+    private final ZonalData<Scalable> zonalScalable;
+
+    ScalableGeneratorConnector(ZonalData<Scalable> zonalScalable) {
+        this.zonalScalable = zonalScalable;
+    }
+
+    void prepareForScaling(Network network) throws ShiftingException {
```

Comment:
do we want to add the countries as parameters of this method for more genericity? (and so it's a bit clearer what's being connected when we call this method)